src/IssueViz.UnitTests/Editor/FilteringTaggerBaseTests.cs

```diff
@@ -0,0 +1,298 @@
+            var inputSpans = new NormalizedSnapshotSpanCollection(snapshot, inputSpan);
+
+            var tagSpan1 = CreateMappingTagSpan(snapshot, new TrackedTag(), new Span(0, 0));
+            var tagSpan2 = CreateMappingTagSpan(snapshot, new TrackedTag(), new Span(1, 5));
+            var tagSpan3 = CreateMappingTagSpan(snapshot, new TrackedTag(), new Span(31, 10));
+            var aggregator = CreateAggregator(tagSpan1, tagSpan2, tagSpan3);
+
+            var testSubject = new TestableFilteringTagger(aggregator, buffer);
+
+            // Act
+            var actual = testSubject.GetTags(inputSpans).ToArray();
+
+            testSubject.FilterCallCount.Should().Be(1);
+            testSubject.LastFilterInput.Should().BeEquivalentTo(tagSpan1, tagSpan2, tagSpan3);
+            actual.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GetTags_HasTagsWithOverlappingSpans_CreatesExpectedTags()
+        {
+            var buffer = CreateBufferWithSnapshot(length: 100);
+            var snapshot = buffer.CurrentSnapshot;
+
+            var inputSpan = new Span(20, 10);
+            var inputSpans = new NormalizedSnapshotSpanCollection(snapshot, inputSpan);
+
+            var tagSpan1 = CreateMappingTagSpan(snapshot, new TrackedTag(), new Span(0, 0));   // no overlap
+            var tagSpan2 = CreateMappingTagSpan(snapshot, new TrackedTag(), new Span(1, 20));  // overlaps [20, 30]
+            var tagSpan3 = CreateMappingTagSpan(snapshot, new TrackedTag(), new Span(30, 10)); // overlaps [20, 30]
+            var tagSpan4 = CreateMappingTagSpan(snapshot, new TrackedTag(), new Span(31, 1));  // no overlap
+            var aggregator = CreateAggregator(tagSpan1, tagSpan2, tagSpan3, tagSpan4);
+
+            var testSubject = new TestableFilteringTagger(aggregator, buffer);
+
+            // Act
+            var actual = testSubject.GetTags(inputSpans).ToArray();
+
+            testSubject.FilterCallCount.Should().Be(1);
+            testSubject.LastFilterInput.Should().BeEquivalentTo(tagSpan1, tagSpan2, tagSpan3, tagSpan4);
+            actual.Select(tagSpan => tagSpan.Tag.SuppliedTrackedTag).Should().BeEquivalentTo(tagSpan2.Tag, tagSpan3.Tag);
+        }
+
+        #region Supporting types
+
+        public class TrackedTag : ITag // needs to be public to be mockable
+        {
+            private static int instanceCounter;
+            public int Id { get; } = ++instanceCounter;
+        }
+
+        private class ProducedTag : ITag
+        {
+            public ProducedTag(TrackedTag trackedTag, NormalizedSnapshotSpanCollection spans)
+            {
+                SuppliedTrackedTag = trackedTag;
+                SuppliedSpans = spans;
+            }
+
+            public TrackedTag SuppliedTrackedTag { get; }
+            public NormalizedSnapshotSpanCollection SuppliedSpans { get; }
+        }
+
+        private class TestableFilteringTagger : FilteringTaggerBase<TrackedTag, ProducedTag>
```

Comment:
This class exists so it would be easier to test the filtering functionality of GetTags. If GetTags were a service that accepts Filter callback, it would've been much easier to test and maintain.

Reply:
This class exists so all of the functionality of the abstract class can be tested i.e. event registration and cleanup, event handling, editor notification, filtering and new tag creation,